homeassistant/components/omnilogic/sensor.py

```diff
@@ -0,0 +1,484 @@
+        super().__init__(
+            coordinator=coordinator,
+            kind=kind,
+            name=name,
+            device_class=device_class,
+            icon=icon,
+            unit=unit,
+            item_id=item_id,
+            state_key=state_key,
+        )
+
+    @property
+    def state(self):
+        """Return the state for the salt level sensor."""
+
+        salt_return = self.coordinator.data[self._item_id].get(self._state_key)
+        unit_of_measurement = CONCENTRATION_PARTS_PER_MILLION
+
+        if self._unit_type == "Metric":
+            salt_return = round(salt_return / 1000, 2)
+            unit_of_measurement = f"{MASS_GRAMS}/{VOLUME_LITERS}"
+
+        self._state = salt_return
+        self._unit = unit_of_measurement
+
+        return self._state
+
+
+class OmniLogicChlorinatorSensor(OmnilogicSensor):
+    """Define an OmniLogic Chlorinator Sensor."""
+
+    def __init__(
+        self,
+        coordinator: OmniLogicUpdateCoordinator,
+        state_key: str,
+        name: str,
+        kind: str,
+        item_id: tuple,
+        device_class: str,
+        icon: str,
+        unit: str,
+    ):
+        """Initialize the sensor."""
+        super().__init__(
+            coordinator=coordinator,
+            kind=kind,
+            name=name,
+            device_class=device_class,
+            icon=icon,
+            unit=unit,
+            item_id=item_id,
+            state_key=state_key,
+        )
+
+    @property
+    def state(self):
+        """Return the state for the chlorinator sensor."""
+        state = self.coordinator.data[self._item_id].get(self._state_key)
+        operating_mode = self.coordinator.data[self._item_id].get("operatingMode")
+
+        if operating_mode == "1":
+            self._state = state
+            self._unit = PERCENTAGE
+        elif operating_mode == "2":
+            self._unit = None
+            if state == "100":
+                self._state = "on"
+            else:
+                self._state = "off"
+
+        return self._state
+
+
+class OmniLogicPHSensor(OmnilogicSensor):
+    """Define an OmniLogic pH Sensor."""
+
+    def __init__(
+        self,
+        coordinator: OmniLogicUpdateCoordinator,
+        state_key: str,
+        name: str,
+        kind: str,
+        item_id: tuple,
+        device_class: str,
+        icon: str,
+        unit: str,
+    ):
+        """Initialize the sensor."""
+        super().__init__(
+            coordinator=coordinator,
+            kind=kind,
+            name=name,
+            device_class=device_class,
+            icon=icon,
+            unit=unit,
+            item_id=item_id,
+            state_key=state_key,
+        )
+
+    @property
+    def state(self):
+        """Return the state for the pH sensor."""
+
+        ph_state = self.coordinator.data[self._item_id].get(self._state_key)
+
+        if ph_state == 0:
+            ph_state = None
+
+        self._state = ph_state
+        self._unit = "pH"
```

Comment:
The unit should be set in the entity settings.

Reply:
Addressed